src/cmd2.cc

```diff
@@ -556,58 +545,9 @@ void do_cmd_go_down(void)
 
 		/* Leaving */
 		p_ptr->leaving = TRUE;
-
-		if (!fall_trap)
-		{
```

Comment:
Hm? Are you seeing actual *shafts* being generated? (Rather than just stairs)

AFAICT this is the *only* place in the code where create_up_shaft is ever set to TRUE.